docs/_tooling/extensions/score_metamodel/checks/id_format_and_length.py

```diff
@@ -0,0 +1,49 @@
+# *******************************************************************************
+# Copyright (c) 2025 Contributors to the Eclipse Foundation
+#
+# See the NOTICE file(s) distributed with this work for additional
+# information regarding copyright ownership.
+#
+# This program and the accompanying materials are made available under the
+# terms of the Apache License Version 2.0 which is available at
+# https://www.apache.org/licenses/LICENSE-2.0
+#
+# SPDX-License-Identifier: Apache-2.0
+# *******************************************************************************
+import os
+
+from sphinx_needs.data import NeedsInfoType
+
+from score_metamodel import CheckLogger, local_check
+
+
+# req-Id: TOOL_REQ__toolchain_sphinx_needs_build__requirement_attributes_uid
+@local_check
+def check_id_format(need: NeedsInfoType, log: CheckLogger):
+    """
+    Checking if the title, directory and feature are included in the requirement id or not.
+    ---
+    """
+    # Split the string by underscores
+    parts = need["id"].split("__")
+
+    if need["type"].startswith(("gd_", "wf_", "wp_")):
+        if len(parts) != 2 and len(parts) != 2:
+            msg = "expected to consisting of one of these 2 formats:`<Req Type>__<Abbreviations>` or `<Req Type>__<Abbreviations>__<Architectural Element>`."
+            log.warning_for_option(need, "id", msg)
+    else:
+        if len(parts) == 3:
```

Comment:
`== 3` is the good case?!
```suggestion
        if len(parts) != 3:
```

Reply:
Am I wrong here? If not, can you please have a deeper look why the tests are passing.

Reply:
Fixed. Please resolve.